File: src/Distance of nearest cell having 1/Code/solution.cpp
#include <bits/stdc++.h>
using namespace std;

class Solution
{
public:
    vector<vector<int>> nearest(vector<vector<int>> &grid)
    {
        int m = grid.size();
        if (m == 0)
            return {};
        int n = grid[0].size();
        vector<vector<int>> dist(m, vector<int>(n, -1));
        queue<pair<int, int>> q;

        // Initialize queue with all 1 cells
        for (int i = 0; i < m; ++i)
        {
            for (int j = 0; j < n; ++j)
            {
                if (grid[i][j] == 1)
                {
                    dist[i][j] = 0;
                    q.push({i, j});
                }
            }
        }

        // Directions (up, down, left, right)
        int dr[4] = {-1, 1, 0, 0};
        int dc[4] = {0, 0, -1, 1};

        // BFS
        while (!q.empty())
        {
            auto [r, c] = q.front();
            q.pop();
            for (int k = 0; k < 4; ++k)
            {
                int nr = r + dr[k];
                int nc = c + dc[k];
                if (nr >= 0 && nr < m && nc >= 0 && nc < n && dist[nr][nc] == -1)
                {
                    dist[nr][nc] = dist[r][c] + 1;
                    q.push({nr, nc});
                }
            }
        }

        return dist;
    }
};

File: src/Distance of nearest cell having 1/Code/solution.py
from collections import deque

class Solution:
	def nearest(self, grid):
		m = len(grid)
		if m == 0:
			return []
		n = len(grid[0])

		dist = [[-1] * n for _ in range(m)]
		q = deque()

		# Initialize all 1s with distance 0 and add them to queue
		for i in range(m):
			for j in range(n):
				if grid[i][j] == 1:
					dist[i][j] = 0
					q.append((i, j))

		# 4-directional moves
		dirs = [(-1,0), (1,0), (0,-1), (0,1)]

		# BFS
		while q:
			r, c = q.popleft()
			for dr, dc in dirs:
				nr, nc = r + dr, c + dc
				if 0 <= nr < m and 0 <= nc < n and dist[nr][nc] == -1:
					dist[nr][nc] = dist[r][c] + 1
					q.append((nr, nc))

		return dist
